[Joseph Nathanael] iP

JosephN37 opened the pull request:
****Duke bot****

> Dukebot is good! - Everyone

Dukebot helps you in remembering things (mostly your tasks for the day)

Dukebot is:
- Easy to use
- Super fast and convenient

All you need to do is:
1. Download it [here](https://github.com/JosephN37/ip)
2. Add your tasks
3. let Duke remember them for you

Features:
- [x] Managing tasks 
- [ ]  Managing deadlines (coming soon!)
- [ ]  Reminders (coming soon!)

If you Java programmer, you can use it to practice Java too. Here's the main method:

    public static void main(String[] args) {

        Duke dukeBot =new Duke("data/Duke.txt");

        dukeBot.run();

    }

Commit:
Add Gradle support

JosephN37 committed:
Greet, Echo, Exit feature ()

JosephN37 committed:
Add, list feature

JosephN37 committed:
Mark as Done feature

JosephN37 committed:
Todos, Events, Deadlines features

JosephN37 committed:
TextUITesting - not done yet, can't seem to run runtest.bat file

JosephN37 committed:
Handle errors feature

JosephN37 committed:
Delete feature

JosephN37 committed:
refactor main into different functions

JosephN37 committed:
Merge pull request #1 from JosephN37/remodelBranch

refactor main into different functions

JosephN37 committed:
Comments added

JosephN37 committed:
add Save feature

JosephN37 committed:
Dates and Times feature

JosephN37 committed:
add Dates and Times format

JosephN37 committed:
Merge pull request #2 from JosephN37/branch-level-7

Branch level 7

JosephN37 committed:
Merge pull request #3 from JosephN37/branch-level-8

add Dates and Times format

JosephN37 committed:
refactor

JosephN37 committed:
refactor and more OOP

JosephN37 committed:
Merge branch 'master' of https://github.com/JosephN37/ip

JosephN37 committed:
Implement packages

JosephN37 committed:
Implement JUnit test

JosephN37 committed:
Add jar file

JosephN37 committed:
add JavaDoc

JosephN37 committed:
comply with coding standards

JosephN37 committed:
add Find feature

JosephN37 committed:
add javadoc

JosephN37 committed:
adhere to coding standards

JosephN37 committed:
Merge pull request #4 from JosephN37/branch-A-JavaDoc

add javadoc

JosephN37 committed:
Merge pull request #5 from JosephN37/branch-A-CodingStandard

adhere to coding standards

JosephN37 committed:
implement find feature

JosephN37 committed:
Merge pull request #6 from JosephN37/branch-Level-9

implement find feature

JosephN37 committed:
Merge branch 'add-gradle-support' of https://github.com/JosephN37/ip into add-gradle-support

mweeruien commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,49 @@
+     */
+    public Deadline(String description, String s) {
+        super(description);
+        this.Dtime = s; // accept "yyyy-mm-dd kkm" format
+        this.localDtime = LocalDateTime.parse(s.substring(1), DateTimeFormatter.ofPattern("uuuu-MM-dd kkmm"));
```
Perhaps it might be better if the "Dtime" variable could be named more clearly as it may not be immediately obvious what the "D" stands for?

mweeruien commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,52 @@
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("uuuu-MM-dd kkmm");
+        String formatDateTime = this.localDtime.format(formatter);
+
+        if(this.isCompleted()) {
```
A space could be left after `if` and before `(this.isCompleted())` to adhere to the coding standards provided.

mweeruien commented on src/main/java/duke/FileUpdater.java:
```diff
@@ -0,0 +1,120 @@
+     * method to rewrite a file to sync the file with the updated tasklist
+     */
+    public void updateListFile(){
+        String s ="";
+        for(int i = 0; i < lst.size(); i++){
```
Perhaps more spaces could be added to adhere to the coding standard? (Namely, before `""` and after `for`.)

mweeruien replied:
There are other instances in your code where it could be better if a space is left as well, maybe another check can be performed?

mweeruien commented on src/main/java/duke/InputChecker.java:
```diff
@@ -0,0 +1,47 @@
+
+        if(s.equals("bye") || s.equals("list")){ //does not throw error if input is bye or list
+
+        } else if(s.equals("done")){
+            throw new DukeException("Please specify what you are done with!");
+        } else if(s.equals("deadline") || s.equals("event") || s.equals("todo")){
+            throw new DukeException("The description of " + s + " cannot be empty!");
+        } else if(s.length() > 4 && s.substring(0,4).equals("todo")){
+
+        } else if(s.length() > 8 && s.substring(0,8).equals("deadline")){
+
+        } else if(s.length() > 5 && s.substring(0,5).equals("event")){
+
+        } else if (s.contains("done") && s.length() > 4){
+
+        } else if (s.equals("delete")) {
+            throw new DukeException("Please specify what you are deleting!");
+        } else if (s.contains("delete")) {
+
+        } else if (s.contains("find") && s.length() > 4){
+
+        } else if (s.equals("find")){
```
Perhaps there might be a better way to organize this? Additionally, it could be better to leave appropriate line breaks and spaces.

mweeruien commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,154 @@
+    }
+
+    /**
+     * method to retrieve the arraylist in a tasklist
```
The Javadocs here could be improved upon to better adhere to the standards provided. An example can be found [here](https://nus-cs2103-ay2122s1.github.io/website/book/documentation/tools/javaDoc/what/), where the description for functions start with verbs such as "Returns" or "Calculates". The Javadocs of other methods has similar areas for improvement, so perhaps it might be better if they were checked?

zhing22 replied:
I agree with mweeruien, in this case perhaps it could be 
```suggestion
     * Getter method to retrieve the arraylist in a tasklist. 
     or 
     * Returns the ArrayList object stored in a tasklist. 
```
Remember to capitalise the first word and add commas to end of sentences.

mweeruien reviewed (commented):
Overall, a good job on your iP, but there are areas for improvement with regards to leaving spaces, line breaks, and in the writing of Javadocs.

zhing22 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,45 @@
+        tasks = new TaskList();
+        ui = new Ui(tasks);
+        try {
+            File f = new File(filePath);
+            this.fileUpdater = new FileUpdater(f, tasks);
```
Perhaps it is better to use a word for variable names (like ```file``` instead of ```f```). I noticed the same issue in several other places too (and also for parameters for some of your methods eg. ```String s``` can be something like ```String input```).

zhing22 commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,7 @@
+package duke;
+
+public class DukeException extends Exception{
+    public DukeException(String errorMessage){
```
Some of your methods are missing spacings before the ```{```. 
```suggestion
public class DukeException extends Exception {
    public DukeException(String errorMessage) {
```
Perhaps you might want to look at the same issue in several other places too

zhing22 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,154 @@
+    public void addDeadline(String s, boolean isCompleted){
+        Deadline d = new Deadline(s.substring(9, s.indexOf("/")),
+                s.substring(s.indexOf("/by") + 3));
+        if(isCompleted)d.completed();
```
Perhaps it is better to put conditional statements in a new line with braces (even if its just one line)
```suggestion
        if (isCompleted) {
            d.completed();
        }
```

zhing22 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,154 @@
+     * @param order
+     */
+    public void deleteTask(int order){
+        System.out.println("Noted! I have removed this task: \n" + lst.get(order-1).toString());
+        lst.remove(order-1);
```
```suggestion
        System.out.println("Noted! I have removed this task: \n" + lst.get(order - 1).toString());
        lst.remove(order - 1);
```

zhing22 reviewed (commented):
Overall nice code! I like how your class and method namings are on point👍 But you might want to take note of tiny details to make your code look cleaner. It would be good to practice consciously and build a habit so it would be easier for you next time (don't have to keep checking back)

JosephN37 committed:
add GUI

JosephN37 committed:
fix gradle run

JosephN37 committed:
Create fat JAR file

JosephN37 committed:
Include assertions in respond(String s) method in GUI class

No assertions are implemented

Assertions are useful in checking whether the user enters the correct
value, for example "delete -1" or "done -2"

JosephN37 committed:
Improve the code to follow coding standard

Coding standards not followed

Change the naming of parameters and variables to enhance readability and
delete dead codes which are not used anymore

JosephN37 committed:
Improve the code to follow coding standard

Coding standards not followed

Change the naming of parameters and variables to enhance readability and
delete dead codes which are not used anymore

JosephN37 committed:
Merge pull request #7 from JosephN37/branch-A-CodeQuality

Branch a code quality

JosephN37 committed:
Merge branch 'master' into branch-A-Assertions

JosephN37 committed:
Merge pull request #8 from JosephN37/branch-A-Assertions

Include assertions in respond(String s) method in GUI class

JosephN37 committed:
fix method comments

JosephN37 committed:
add sorting feature

JosephN37 committed:
fix minHeight of dialogue box to display all tasks in the list

JosephN37 committed:
Merge branch 'branch-C-Sort'

JosephN37 committed:
modify greet funtion to include sort

JosephN37 committed:
include screenshot of duke ui

JosephN37 committed:
rename ui screenshot

JosephN37 committed:
update README.md to include user guide

JosephN37 committed:
initialize file when input file to duke does not exist

JosephN37 committed:
add screenshot
